yt_dlp/extractor/ted.py

```diff
@@ -149,84 +90,76 @@ def _real_extract(self, url):
-        for entry in re.findall(r'(?s)<[^>]+data-ga-context=["\']playlist["\'][^>]*>', webpage):
-            attrs = extract_attributes(entry)
-            entry_url = compat_urlparse.urljoin(url, attrs['href'])
-            playlist_entries.append(self.url_result(entry_url, self.ie_key()))
-
-        final_url = self._og_search_url(webpage, fatal=False)
-        playlist_id = (
-            re.match(self._VALID_URL, final_url).group('playlist_id')
-            if final_url else None)
+        playlist = try_get(json, lambda x: x['props']['pageProps']['playlist'], dict)
+        playlist_id = playlist.get('id')
+        playlist_entries = self._parse_playlist(playlist)
 
         return self.playlist_result(
             playlist_entries, playlist_id=playlist_id,
             playlist_title=self._og_search_title(webpage, fatal=False),
             playlist_description=self._og_search_description(webpage))
 
-    def _talk_info(self, url, video_name):
-        webpage = self._download_webpage(url, video_name)
+    def _series_videos_info(self, url, name, season):
+        '''Returns the videos of the series playlist'''
+        webpage = self._download_webpage(url, name, 'Downloading series webpage')
+        json = self._extract_info(webpage, 'json')
+        series = try_get(json, lambda x: x['props']['pageProps']['series'])
+        series_id = series.get('id')
 
-        info = self._extract_info(webpage)
+        seasonlist = try_get(json, lambda x: x['props']['pageProps']['seasons'], list)
 
-        data = try_get(info, lambda x: x['__INITIAL_DATA__'], dict) or info
-        talk_info = data['talks'][0]
-
-        title = talk_info['title'].strip()
-
-        downloads = talk_info.get('downloads') or {}
-        native_downloads = downloads.get('nativeDownloads') or talk_info.get('nativeDownloads') or {}
-
-        formats = [{
-            'url': format_url,
-            'format_id': format_id,
-        } for (format_id, format_url) in native_downloads.items() if format_url is not None]
-
-        subtitled_downloads = downloads.get('subtitledDownloads') or {}
-        for lang, subtitled_download in subtitled_downloads.items():
-            for q in self._NATIVE_FORMATS:
-                q_url = subtitled_download.get(q)
-                if not q_url:
-                    continue
-                formats.append({
-                    'url': q_url,
-                    'format_id': '%s-%s' % (q, lang),
-                    'language': lang,
-                })
-
-        if formats:
-            for f in formats:
-                finfo = self._NATIVE_FORMATS.get(f['format_id'].split('-')[0])
-                if finfo:
-                    f.update(finfo)
-
-        player_talk = talk_info['player_talks'][0]
+        playlist_entries = []
+        if season:
+            for s in [playlist for playlist in seasonlist if playlist['seasonNumber'] == season]:
+                playlist_entries.extend(self._parse_playlist(s))
```

Comment:
```suggestion
            playlist_entries = [self._parse_playlist(s) for s in seasonlist if s.get('seasonNumber') == season]
```

Reply:
I am getting errors with your suggested change.
Still haven't got my head around the Python for loop short forms, please check if this change below is acceptable.

https://github.com/trassshhub/yt-dlp/blob/6ea4d15a3d79dd9f504a8ca242c8c40a0cf2c080/yt_dlp/extractor/ted.py#L114-L118

Reply:
let it be as-is for now